File: java/abc206/B.java
import java.util.*;

public class B {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int N = sc.nextInt();
    // i**2+i-2n >= 0
    double ans = (-1.0D + Math.sqrt(1.0D + 8.0D * (double) N)) / 2.0D;
    System.out.println((int) Math.ceil(ans));
  }

}

File: rust/abc206/src/b.rs
use proconio::input;
// use proconio::marker::Chars;
// use std::collections::HashMap;
use std::io::{stdout, BufWriter, Write};

pub fn main() {
  let out = stdout();
  let mut out = BufWriter::new(out.lock());

  input! {n: u64}
  // total = i(i+1)/2
  // min { i: i(i+1) >= 2n }
  // i.pow(2)+1-2n >= 0
  // i = (-1 + sqrt(1+8n))/2
  let ans: f64 = (-1.0 + (1.0f64 + 8.0 * n as f64).sqrt()) / 2.0;
  writeln!(out, "{}", ans.ceil() as u64).unwrap();
}
